src/utils/cosmosdb_model_versioned.ts

```diff
@@ -166,6 +172,66 @@ export abstract class CosmosdbModelVersioned<
     );
   }
 
+  /**
+   * Add the given ttl at all the version of the document identified by
+   * the searchKey
+   */
+
+  public updateTTLForAllVersion(
+    searchKey: DocumentSearchKey<TR, ModelIdKey, PartitionKey>,
+    ttl: NonNegativeNumber
+  ): TE.TaskEither<CosmosErrors, ReadonlyArray<TR>> {
+    const partitionKey = searchKey.length === 1 ? searchKey[0] : searchKey[1];
+    return pipe(
+      this.findAllVersionsByPartitionKey(searchKey),
+      TE.map(RA.rights),
+      TE.map(
+        RA.map(document =>
+          super.updateTTL(
+            ([document.id, partitionKey] as unknown) as DocumentSearchKey<
+              TR,
+              CosmosDocumentIdKey,
+              PartitionKey
+            >,
+            ttl
+          )
+        )
+      ),
+      TE.chain(RA.sequence(TE.ApplicativeSeq)),
```

Comment:
What if some of this operation fail permanently (over the internal cosmos SDK retry)? We could have an inconsistent state when some version are updated and other aren't. Can we consider a batch operation instead? Maybe we can't, seems that transactional batch are not yet supported from the node SDK. https://github.com/Azure/azure-sdk-for-js/issues/13368

Reply:
Unfortunately it seems there's no way to do it with node SDK. 
We need to be careful on monitoring, and fix it manually I think.
@AleDore @fabriziopapi

Reply:
I think we can try with latest sdk version if needed. It seems that transactional support has been merged https://github.com/Azure/azure-sdk-for-js/pull/16838